### ISSUE TITLE
Generating a new pdf document fails when there is an indexed png without transparency

### ISSUE THREAD

**Issue by kriztynna** [prawnpdf/prawn#937]
Generating a pdf using indexed png images with transparency works, but if the images do not have transparency, it fails. Submitting a PR shortly with the fix and related spec.

```
#<NoMethodError: undefined method `[]' for nil:NilClass> with backtrace:
         # ./lib/prawn/images/png.rb:301:in `block in generate_alpha_channel'
         # ./lib/prawn/images/png.rb:295:in `upto'
         # ./lib/prawn/images/png.rb:295:in `generate_alpha_channel'
         # ./lib/prawn/images/png.rb:112:in `split_alpha_channel!'
         # ./lib/prawn/images/png.rb:144:in `build_pdf_object'
         # ./lib/prawn/images.rb:93:in `build_image_object'
         # ./lib/prawn/images.rb:66:in `image'
         # ./spec/acceptance/png_spec.rb:31:in `block (6 levels) in <top (required)>'
         # ./lib/prawn/document.rb:226:in `instance_eval'
         # ./lib/prawn/document.rb:226:in `initialize'
         # ./lib/prawn/document.rb:142:in `new'
         # ./lib/prawn/document.rb:142:in `generate'
         # ./spec/acceptance/png_spec.rb:23:in `block (5 levels) in <top (required)>'
         # ./spec/acceptance/png_spec.rb:22:in `block (4 levels) in <top (required)>'
```

**Comment by cheba** [comment]
It looks like it's the same issue as #933. Is this the case?

**Comment by kriztynna** [comment]
@cheba Yes same issue, different PR.

**Comment by cheba** [comment]
@kriztynna Good. Looking forward to it.

**Comment by kriztynna** [comment]
Resolved by #938
